### PR TITLE
Make `file_options` an opaque object for ABI compatibility

### PR DESCRIPTION
**Pull request by umangyadav**

### CODE REVIEW

**Review by kahmed10 — APPROVED** [review]

File: src/api/api.cpp [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#include <migraphx/operation.hpp>
```
Comment (scxiao) [inline src/api/api.cpp:1]: Do we need this include here?

File: src/api/api.cpp [hunk 2]
```diff
@@ -320,6 +316,16 @@ struct migraphx_onnx_options
     migraphx::onnx_options object;
 };
 
+extern "C" struct migraphx_file_options;
```
Comment (scxiao) [inline src/api/api.cpp:319]: How about the rename this to serialization_option? @pfultz2

File: src/api/api.cpp (hunk 1 above)
Comment (umangyadav) [reply src/api/api.cpp:1]: It is auto-generated from `tools/.../api.cpp`  It is needed.

File: src/api/api.cpp (hunk 2 above)
Comment (pfultz2) [reply src/api/api.cpp:319]: Lets just keep the name the same.

**Review by pfultz2 — APPROVED** [review]
I dont think we need to worry about the code coverage here.

**Review by scxiao — APPROVED** [review]

**Pull request merged**